### PR TITLE
[MRG+1] Adds docstrings to demo_curvelinear_grid.py and demo_curvelinear_grid…

### PR DESCRIPTION
**Pull request by trpham**

### CODE REVIEW

**Comment by tacaswell** [comment]
It looks like some extraneous files got committed.  Can you remove those from the git history and then force push?


-----

```sh
# make sure you are on your working branch
$ git checkout 7206-MEP12-compliant-Plots
# do this a bunch of times 
$ git rm file_name
# ammend the last commit
$ git commit --amend 
# _FORCE_ push to github.
# this is dangerous, use with caution
$ git push trpham 7206-MEP12-compliant-Plots --force
```

**Comment by NelleV** [comment]
Hi @trpham 
In addition to @tacaswell's comment to remove the ipython notebook files, can you be a bit more specific in the docstring?
Specifically, I am looking for a docstring with a short title and a description, formatted as is:

"""
title

description
"""

File: examples/axisartist/demo_curvelinear_grid.py [hunk 1]
```diff
@@ -1,3 +1,17 @@
+"""
+Demo of defining curvilinear coordinate using GridHelperCurveLinear and Transformation.
```
Comment (NelleV) [inline examples/axisartist/demo_curvelinear_grid.py]: The docstring looks good overall, but here is a couple of improvements you should make:

* Let's try to make the title shorter: something like "Curvilinear coordinates" would be better.
* Add a sentence at the beginning of the paragraph to introduce it: This examples illustrates how to define curvilinar coordinates using GridHelperCurveLinear and Transformation".

Comment (QuLogic) [inline examples/axisartist/demo_curvelinear_grid.py]: Please wrap at 79 characters.

File: examples/axisartist/demo_curvelinear_grid.py [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+In curvelinear_test1: Use GridHelper class
+    * User provides two transformtions function between curved coordinate and rectlinear coordinate
```
Comment (QuLogic) [inline examples/axisartist/demo_curvelinear_grid.py]: transformations; rectilinear.

File: examples/axisartist/demo_curvelinear_grid.py [hunk 3]
```diff
@@ -1,3 +1,17 @@
...
+    * Calling GridHelperCurveLinear on those two functions.
+    * Setting a subplot positioned by the given gird definition
```
Comment (QuLogic) [inline examples/axisartist/demo_curvelinear_grid.py]: grid

File: examples/axisartist/demo_curvelinear_grid.py [hunk 4]
```diff
@@ -1,3 +1,17 @@
...
+In curvelinear_test2: Use Affine2D transform
+    * Calling PolerTransform
```
Comment (QuLogic) [inline examples/axisartist/demo_curvelinear_grid.py]: PolarTransform

File: examples/axisartist/demo_curvelinear_grid.py [hunk 5]
```diff
@@ -1,3 +1,17 @@
+"""
+Demo of defining curvilinear coordinate using GridHelperCurveLinear and Transformation.
+In curvelinear_test1: Use GridHelper class
+    * User provides two transformtions function between curved coordinate and rectlinear coordinate
+    * Calling GridHelperCurveLinear on those two functions.
+    * Setting a subplot positioned by the given gird definition
+In curvelinear_test2: Use Affine2D transform
+    * Calling PolerTransform 
+    * Setting the range of coordinate using Extream Finder
```
Comment (QuLogic) [inline examples/axisartist/demo_curvelinear_grid.py]: coordinates; Extreme

File: examples/axisartist/demo_curvelinear_grid.py [hunk 6]
```diff
@@ -1,3 +1,17 @@
...
+    * Locate the coordinate with LocatorDMS and format them with FormatterDMS
+    * Calling GridHelperCurveLinear
+    * Setting a subplot positioned by the given gird definition
```
Comment (QuLogic) [inline examples/axisartist/demo_curvelinear_grid.py]: grid

**Comment by NelleV** [comment]
Hi @trpham 
The docstring is not formatted as I'd like: we need both a title and a description for each plot. The title can be much shorter and to the point. I realize that those examples are very complicated to understand, and the function they use often undocumented.
Here is a suggestion:

"""
Custom grid and ticklines.

This example demonstrates how to use GridHelperCurveLinear to define custom grids and ticklines by applying a transformation on the grid. This can be used, as showcase on the second plot, to create polar projections in a rectangular box.
"""

You can leave the second example as is. I think it needs to be removed or updated to be interesting.

**Comment by QuLogic** [comment]
The first commit has a confusing commit message; no files were removed at the time.

**Comment by NelleV** [comment]
@tacaswell Why has this been tagged for 2.0? This is not a bug fix.

**Comment by QuLogic** [comment]
Backported to v2.x via bceea0ca89cab3b3b7f782ac468b33364b2778c4.

**Pull request merged**